sql/create.go

```diff
@@ -142,6 +142,8 @@ func (p *planner) CreateIndex(n *parser.CreateIndex) (planNode, error) {
 		return nil, err
 	}
 
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
I agree with @tschottdorf that this seems to make more sense in the sql session (and a field of `EndTransactionRequest`) than in the `Transaction` itself. The sql session is persisted through similar mechanisms to account for multi-statement transactions, and the sql planner could set the field in `EndTransactionRequest` based on what's in the session when it commits.